velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,333 @@
+ * data representation,
+ * expression evaluation,
+ * aggregations,
+ * sorting,
+ * joins,
+ * accessing data via connectors.
+
+Note: Velox doesn't provide a SQL interface, but in this tutorial to keep
+things simple we are going to use SQL and utilities that were designed
+to simplify test development. These utilities use `DuckDB <https://duckdb.org/>`_
+to parse SQL.
+
+The code used in this tutorial is available in `velox/exec/tests/VeloxIn10MinDemo.cpp <https://github.com/facebookincubator/velox/blob/main/velox/exec/tests/VeloxIn10MinDemo.cpp>`_.
+
+Let’s get started.
+
+Vectors
+-------
+
+Velox uses columnar representation of the data similar to Arrow called vectors.
+Let’s create two vectors to store 64-bit integers.
+
+.. code-block:: c++
+
+    auto a = makeFlatVector<int64_t>({0, 1, 2, 3, 4, 5});
+    auto b = makeFlatVector<int64_t>({0, 5, 10, 15, 20, 25});
+
+And let’s print out the contents of these vectors:
+
+.. code-block:: c++
+
+    auto data = makeRowVector({"a", "b"}, {a, b});
+
+    std::cout << data->toString() << std::endl;
+    std::cout << data->toString(0, data->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT>: 6 elements, no nulls]
+
+    0: {0, 0}
+    1: {1, 5}
+    2: {2, 10}
+    3: {3, 15}
+    4: {4, 20}
+    5: {5, 25}
+
+Check out :doc:`Vectors <develop/vectors>` guide to learn more about
+Velox vectors.
+
+Expressions
+-----------
+
+Now, let’s compute a sum of 'a' and 'b' by evaluating 'a + b' expression.
+
+First we use DuckDB SQL parser to parse the expression into a fully typed
+expression tree. Then, we use Velox APIs to compile the expression it into
+an executable ExprSet. Check out the implementation of the compileExpression
+helper method in VeloxIn10MinDemo.cpp for details.
+
+.. code-block:: c++
+
+    auto exprSet = compileExpression("a + b", asRowType(data->type()));
+
+Let's print out the compiled expression:
+
+.. code-block::
+
+    > plus(a, b)
+
+Now we are ready to evaluate the expression on a batch of data.
+
+.. code-block:: c++
+
+    auto c = evaluate(*exprSet, data);
+
+And here are the results:
+
+.. code-block:: c++
+
+    auto abc = makeRowVector({"a", "b", "c"}, {a, b, c});
+
+    std::cout << abc->toString() << std::endl;
+    std::cout << abc->toString(0, c->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT,c:BIGINT>: 6 elements, no nulls]
+    0: {0, 0, 0}
+    1: {1, 5, 6}
+    2: {2, 10, 12}
+    3: {3, 15, 18}
+    4: {4, 20, 24}
+    5: {5, 25, 30}
+
+Let's try a slightly more complex expression: `2 * a + b % 3`.
+
+.. code-block:: c++
+
+    exprSet = compileExpression("2 * a + b % 3", asRowType(data->type()));
+    std::cout << exprSet->toString() << std::endl;
+
+.. code-block::
+
+    > plus(multiply(2:BIGINT, a), mod(b, 3:BIGINT))
+
+.. code-block:: c++
+
+    auto d = evaluate(*exprSet, data);
+
+    auto abcd = makeRowVector({"a", "b", "c", "d"}, {a, b, c, d});
```

Comment:
I think we don't need to print out the "c" column here.

Reply:
Ok.

Reply:
@kagamiori Updated.